managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -3663,7 +3663,7 @@ Pair<PositionImpl, Long> getFirstPositionAndCounter() {
 
     public void activateCursor(ManagedCursor cursor) {
         synchronized (activeCursors) {
-            if (activeCursors.get(cursor.getName()) == null) {
+            if (!cursor.isActive()) {
```

Comment:
If change like this, I think it's better like 
```
if (!cursor.isActive()) {
   Position positionForOrdering = config.isCacheEvictionByMarkDeletedPosition()
                        ? cursor.getMarkDeletedPosition()
                        : cursor.getReadPosition();
     if (positionForOrdering == null) {
             positionForOrdering = PositionImpl.EARLIEST;
     }
     synchronized (activeCursors) {
          activeCursors.add(cursor, positionForOrdering);
     }
}          
```

Reply:
Just for reference, this change ultimately relies on a call to this method:

https://github.com/apache/pulsar/blob/3bb93cb95e6c5290f59d6b7b4a57ec8296691569/managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java#L3695-L3697

We don't want to change the scope of the `synchronized` block, as far as I can tell.